tests/src/PatternAssertions/CardAssert.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_whitelabel\PatternAssertions;
+
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Assertions for the card pattern.
+ *
+ * @see ./templates/patterns/card/card.ui_patterns.yml
+ */
+class CardAssert extends BasePatternAssert {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getAssertions($variant): array {
+    return [
+      'title' => [
+        [$this, 'assertElementText'],
+        '.card-title a span',
+      ],
+      'url' => [
+        [$this, 'assertElementAttribute'],
+        '.card-title a',
+        'href',
+      ],
+      'image' => [
+        [$this, 'assertCardImage'],
+        $variant,
+      ],
+      'description' => [
+        [$this, 'assertElementText'],
+        '.card-text',
+      ],
+      'badges' => [
+        [$this, 'assertBadgesElements'],
+      ],
+      'content' => [
+        [$this, 'assertContent'],
+      ],
+    ];
+  }
+
+  /**
+   * Asserts the image of a card.
+   *
+   * @param array|null $expected_image
+   *   The expected image values.
+   * @param string $variant
+   *   The variant of the pattern being checked.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The DomCrawler where to check the element.
+   */
+  protected function assertCardImage($expected_image, string $variant, Crawler $crawler): void {
+    if ($variant == 'search') {
+      $image_div = $crawler->filter('.mw-listing-img img.card-img-top');
+      self::assertEquals($expected_image['alt'], $image_div->attr('alt'));
+      self::assertStringContainsString($expected_image['src'], $image_div->attr('src'));
+    }
+    else {
+      $image_div = $crawler->filter('div.card img');
+      self::assertEquals($expected_image['alt'], $image_div->attr('alt'));
+      self::assertStringContainsString($expected_image['src'], $image_div->attr('src'));
+    }
+  }
+
+  /**
+   * Asserts the content of a card.
+   *
+   * @param array $expected_items
+   *   The expected item values.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The DomCrawler where to check the element.
+   */
+  protected function assertContent(array $expected_items, Crawler $crawler): void {
+    foreach ($expected_items as $expected_item) {
+      self::assertStringContainsString($expected_item, $crawler->html());
```

Comment:
let's leave a comment here explaining that there is no wrapping element that we can target so we check that the expected items are present